crdt/sequence.py

```diff
@@ -0,0 +1,247 @@
+from functools import cmp_to_key
+from re import S
+import uuid
+from enum import Enum
+
+from crdt.gset import GSet
+from crdt.gcounter import GCounter
+
+class Sequence():
+    """
+    Sequence is a CRDT that represents an ordered set of objects and supports insertion
+    and removal at arbitrary positions in the set.
+    """
+
+    def __init__(self, id=uuid.uuid4()):
+        self.id = id
+        self.operations = GSet()
+        self.clock = GCounter(self.id)
+        self.sequence = []
+
+    def compare_operations(self, a, b):
+        """
+        Compares this Operation to another Operation to determine ordering.
+        """
+        return a.owner.compare(b.owner)
+
+    def index_transform(self, position):
+        """
+        Returns the transformed index, which is the actual index of the object in the
+        sequence, taking into account tombstones.
+        """
```

Comment:
Might also be useful to document the input and output here; something like:

`position`: int
    the place where the user intends to insert, but which does not take into consideration items that have been removed by prior operations (aka tombstones)
`index`: int
     the correct insertion place, taking into account tombstones